test/cli/kit/executor_test.rb

```diff
@@ -18,21 +18,44 @@ def test_nil_log
         exe = Executor.new(log_file: @tf.path)
         out, err = capture_io do
           CLI::UI::StdoutRouter.with_enabled do
-            exe.call(method(:simple_command), 'foo', %w(a b))
+            exe.call(SimpleCommand, 'foo', %w(a b))
           end
         end
         assert_equal(%(foo: ["a", "b"]\n), out)
         assert_empty(err)
       end
 
+      def test_call_with_exception
+        SimpleCommand.any_instance.expects(:call).raises(StandardError)
+        CLI::UI::StdoutRouter.expects(:with_id).yields("12345")
+
+        exe = Executor.new(log_file: @tf.path)
+        out, err = capture_io do
+          assert_raises StandardError do
+            CLI::UI::StdoutRouter.with_enabled do
+              exe.call(SimpleCommand, 'foo', %w(a b))
+            end
+          end
+        end
+
+        assert_equal(<<~EOF, err)
+        This command ran with ID: 12345
+        Please include this information in any issues/report along with relevant logs
+        EOF
+        assert_empty(out)
+      end
+
       def test_command_runs
         out, err = capture_io do
           CLI::UI::StdoutRouter.with_enabled do
-            exe.call(method(:simple_command), 'foo', %w(a b))
```

Comment:
I tested locally:

```rb
out, err = capture_io do
  CLI::UI::StdoutRouter.with_enabled do
    exe.call(SimpleCommand, 'foo', %w(a b))
    exe.call(SimpleCommand, 'foo', %w(a b))
    exe.call(SimpleCommand, 'foo', %w(a b))
  end
end
puts File.read(@tf.path)
```
It prints:
```
foo: ["a", "b"]
foo: ["a", "b"]
foo: ["a", "b"]
[56290] foo: ["a", "b"]
```

i.e., the output accumulates, but not the file logs. Is this expected?
Is it's worth adding a test for that too?

Reply:
I suspect it's weird file closing with the `Tempfile`